r/tests/testthat/test-normalized_matrix.R

```diff
@@ -11,64 +11,113 @@ mock_req <- function(apply_subset = TRUE) {
     body = list(
       subsetBy = subset_by,
       applySubset = apply_subset
-        )
-      )
+    )
+  )
+}
+
+mockery::stub(
+  GetNormalizedExpression,
+  "vroom::vroom_write",
+  INTERNAL_RESULTS_PATH
+)
+
+stubbed_GetNormalizedExpression <- function(req, data) {
+  GetNormalizedExpression(req, data)
```

Comment:
Usually this is done inside the stubbed function definition, like:

```suggestion

stubbed_GetNormalizedExpression <- function(req, data) {
  mockery::stub(
  GetNormalizedExpression,
  "vroom::vroom_write",
  INTERNAL_RESULTS_PATH
)

  GetNormalizedExpression(req, data)
```

Reply:
also, the parameter `INTERNAL_RESULTS_PATH` will not really work. `mockery::stub` takes as arguments the "where" (you want the stubbing to happen), the "what" (you want to stub, i.e. change behaviour) and the "how" (you change the behaviour). 

Usually this means replacing (stubbing) the problematic function call with a function with a behaviour that's nice for testing purposes (a mock function). 

Here it means that you're replacing the `vroom::vroom_write` with the global variable `INTERNAL_RESULTS_PATH`